app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
-          else
-            flash[:error] = t('error_banned', :default => "User could not be permanently suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
-          end
-        elsif params[:admin_action] == 'unsuspend'
-          if @user.suspended?
-            @user.suspended = false
-            @user.suspended_until = nil
-            if @user.save && !@user.suspended? && @user.suspended_until.blank?
-              @user.create_log_item( options = {:action => ArchiveConfig.ACTION_UNSUSPEND, :note => @admin_note, :admin_id => current_admin.id})
-              flash[:notice] = t('success_unsuspend', :default => "Suspension has been lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            else
-              flash[:error] = t('error_unsuspend', :default => "Suspension could not be lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            end
-          else
-            flash[:notice] = t('not_suspended', :default => "User had not been suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_BAN, :note => @admin_note, :admin_id => current_admin.id})
+          success_message << ts('User has been permanently suspended.')
+        end
+
+        # unsuspended suspended user
+        if params[:admin_action] == 'unsuspend' && @user.suspended?
+          @user.suspended = false
+          @user.suspended_until = nil
+          if !@user.suspended && @user.suspended_until.blank?
+            @user.create_log_item( options = {:action => ArchiveConfig.ACTION_UNSUSPEND, :note => @admin_note, :admin_id => current_admin.id})
+            success_message << ts('Suspension has been lifted.')
           end
-        elsif params[:admin_action] == 'unban'
-          if @user.banned?
-            @user.banned = false
-            if @user.save && !@user.banned?
-              @user.create_log_item( options = {:action => ArchiveConfig.ACTION_UNSUSPEND, :note => @admin_note, :admin_id => current_admin.id})
-              flash[:notice] = t('success_unsuspend', :default => "Suspension has been lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            else
-              flash[:error] = t('error_unsuspend', :default => "Suspension could not be lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            end
-          else
-            flash[:notice] = t('not_banned', :default => "User had not been permanently suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
+        end
+
+        # unban banned user
+        if params[:admin_action] == 'unban' && @user.banned?
+          @user.banned = false
+          if !@user.banned?
+            @user.create_log_item( options = {:action => ArchiveConfig.ACTION_UNSUSPEND, :note => @admin_note, :admin_id => current_admin.id})
```

Comment:
Useless assignment to variable - `options`.<br>Line is too long. [142/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.<br>Space inside parentheses detected.